test/code-build-test.js

```diff
@@ -318,6 +318,104 @@ describe("waitForBuildEndTime", () => {
     });
     expect(test).to.equal(buildReplies.pop().builds[0]);
   });
+
+  it("waits after being rate limited and tries again", async function() {
+    this.timeout(45000);
+    const buildID = "buildID";
+    const nullArn =
+      "arn:aws:logs:us-west-2:111122223333:log-group:null:log-stream:null";
+    const cloudWatchLogsArn =
+      "arn:aws:logs:us-west-2:111122223333:log-group:/aws/codebuild/CloudWatchLogGroup:log-stream:1234abcd-12ab-34cd-56ef-1234567890ab";
+
+    const buildReplies = [
+      () => {
+        throw { message: "Rate exceeded" };
+      },
+      { builds: [{ id: buildID, logs: { cloudWatchLogsArn } }] },
+      {
+        builds: [
+          { id: buildID, logs: { cloudWatchLogsArn }, endTime: "endTime" }
+        ]
+      }
+    ];
+
+    const sdk = help(
+      () => {
+        //similar to the ret function in the helper, allows me to throw an error in a function or return a more standard reply
+        let reply = buildReplies.shift();
+
+        if (typeof reply === "function") return reply();
+        return reply;
+      },
+      () => {
+        if (!buildReplies.length) {
+          return { events: [] };
+        }
+
+        return { events: [{ message: "got one" }] };
+      }
+    );
+
+    const test = await waitForBuildEndTime(sdk, {
```

Comment:
now that we have all the values,
use `wait` and `backOff` to keep the test fast.
See the `help` function.